modules/apps/portal-search/portal-search-web/src/test/java/com/liferay/portal/search/web/internal/facet/display/context/UserSearchFacetDisplayContextTest.java

```diff
@@ -172,10 +175,83 @@ public void testOneTermWithPreviousSelection() throws Exception {
 		Assert.assertFalse(userSearchFacetDisplayContext.isRenderNothing());
 	}
 
+	@Test
+	public void testOrderByTermFrequencyAscending() throws Exception {
+		String[] userNames = {"alpha", "bravo", "delta", "charlie"};
+
+		_setUpMultipleTermCollectors(
+			_getTermCollectors(userNames, new int[] {6, 5, 4, 3}));
+
+		UserSearchFacetDisplayContext userSearchFacetDisplayContext =
+			createDisplayContext(StringPool.BLANK, "count:asc");
+
+		String nameFrequencyString = _buildNameFrequencyString(
+			userSearchFacetDisplayContext.getBucketDisplayContexts());
+
+		Assert.assertEquals(
+			"charlie:3|delta:4|bravo:5|alpha:6", nameFrequencyString);
+	}
+
+	@Test
+	public void testOrderByTermFrequencyDescending() throws Exception {
+		String[] userNames = {"alpha", "charlie", "bravo", "delta"};
+
+		_setUpMultipleTermCollectors(
+			_getTermCollectors(userNames, new int[] {3, 4, 5, 6}));
+
+		UserSearchFacetDisplayContext userSearchFacetDisplayContext =
+			createDisplayContext(StringPool.BLANK, "count:desc");
+
+		String nameFrequencyString = _buildNameFrequencyString(
+			userSearchFacetDisplayContext.getBucketDisplayContexts());
+
+		Assert.assertEquals(
+			"delta:6|bravo:5|charlie:4|alpha:3", nameFrequencyString);
+	}
+
+	@Test
+	public void testOrderByTermValueAscending() throws Exception {
+		String[] userNames = {"bravo", "delta", "alpha", "charlie"};
+
+		_setUpMultipleTermCollectors(_getTermCollectors(userNames));
+
+		UserSearchFacetDisplayContext userSearchFacetDisplayContext =
+			createDisplayContext(StringPool.BLANK, "key:asc");
+
+		String nameFrequencyString = _buildNameFrequencyString(
+			userSearchFacetDisplayContext.getBucketDisplayContexts());
+
+		Assert.assertEquals(
+			"alpha:3|bravo:1|charlie:4|delta:2", nameFrequencyString);
+	}
+
+	@Test
+	public void testOrderByTermValueDescending() throws Exception {
+		String[] userNames = {"bravo", "delta", "alpha", "charlie"};
+
+		_setUpMultipleTermCollectors(_getTermCollectors(userNames));
+
+		UserSearchFacetDisplayContext userSearchFacetDisplayContext =
+			createDisplayContext(StringPool.BLANK, "key:desc");
+
+		String nameFrequencyString = _buildNameFrequencyString(
+			userSearchFacetDisplayContext.getBucketDisplayContexts());
+
+		Assert.assertEquals(
+			"delta:2|charlie:4|bravo:1|alpha:3", nameFrequencyString);
+	}
+
 	protected UserSearchFacetDisplayContext createDisplayContext(
 			String paramValue)
 		throws Exception {
 
+		return createDisplayContext(paramValue, "count:desc");
+	}
+
+	protected UserSearchFacetDisplayContext createDisplayContext(
```

Comment:
Brian wants this private (and reordered) https://github.com/brianchandotcom/liferay-portal/commit/eb4a97141e66a5fcb508d10de01ca70fea80c7cc